lib/opal/cache/file_cache.rb

```diff
@@ -78,19 +78,14 @@ def get(key)
         nil
       end
 
-      # This complex piece of code tries to check if we can robustly mkdir_p a directory.
+      # Check if we can robustly mkdir_p a directory.
       def self.dir_writable?(*paths)
-        dir = nil
-        paths = paths.reduce([]) do |a, b|
-          [*a, dir = a.last ? File.expand_path(b, a.last) : b]
+        dir = "/"
+        until paths.empty?
+          dir = File.expand_path(paths.shift, dir)
+          ok = File.directory?(dir) && File.writable?(dir) if File.exist?(dir)
         end
-
-        File.exist?(paths.first) &&
-          paths.reverse.all? do |i|
-            !File.exist?(i) || (File.directory?(i) && File.writable?(i))
-          end
-
-        dir
+        dir if ok
```

Comment:
Nice improvement!
Since `expand_path` accepts `nil` as the second argument it can be skipped, which makes it compatible with windows, also added a check on the first path segment to ensure it exist, it being the "root".

```rb
       def self.dir_writable?(*paths)
         return false unless File.exist?(paths.first)
         
         until paths.empty?
           dir = File.expand_path(paths.shift, dir)
           ok = File.directory?(dir) && File.writable?(dir) if File.exist?(dir)
         end

         dir if ok
       end
```

Reply:
Ah, Windows compatibility, didn't think of that one.

The PR is marked as "changes requested" but you already made the changes, so I'm not sure if you're waiting on something from me?

Reply:
We haven't committed that change to the branch. We did now :) Merging in a moment